### DISCUSSION TITLE
How can I change the from mail name?

### DISCUSSION THREAD

**Discussion by saschabrockel** [healthchecks/healthchecks#1045] [Q&A] [answered]
In my services, you can change the name of your from mail that it is received like what you defined instead of example@org.net.

I thought changing the environment variable `DEFAULT_FROM_EMAIL` to `Healthchecks <example@org.net>` would help but it does not work. In Gmail, the from mail still appears as example@org.net. It normally works for all my other services. How to do it here?

**Comment by cuu508** [comment]
On the hosted service, Healthchecks.io, I use:

```
DEFAULT_FROM_EMAIL = '"Healthchecks.io" <healthchecks.io@healthchecks.io>'
```

–and it works.

One thing to check, after you changed the `DEFAULT_FROM_EMAIL` value, did you restart the `sendalerts` process?

**Reply by saschabrockel** [reply]
No matter what I try, it doesn't work. My unRAID was running the command now with   `-e DEFAULT_FROM_EMAIL='"Healthchecks" <example@org.net>'`

Still not working. Any why do I need to do that with sendalerts when I restart the Docker container already?

My config:

```
docker run
  -d
  --name='healthchecks'
  --net='unraid'
  -e TZ="Europe/Berlin"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="UNRAID-Server"
  -e HOST_CONTAINERNAME="healthchecks"
  -e 'SITE_ROOT'='https://health.example.xyz'
  -e 'SITE_NAME'='Healthchecks'
  -e 'EMAIL_HOST'='ssl.xyz.net'
  -e 'EMAIL_PORT'='587'
  -e 'EMAIL_HOST_USER'='example@org.net'
  -e 'EMAIL_HOST_PASSWORD'='PASSWORD'
  -e 'EMAIL_USE_TLS'='True'
  -e 'SUPERUSER_EMAIL'='example@org.net'
  -e 'SUPERUSER_PASSWORD'='PASSWORD'
  -e 'REGENERATE_SETTINGS'=''
  -e 'ALLOWED_HOSTS'='*'
  -e 'APPRISE_ENABLED'='True'
  -e 'DEBUG'='False'
  -e 'INTEGRATIONS_ALLOW_PRIVATE_IPS'=''
  -e 'PING_EMAIL_DOMAIN'=''
  -e 'SECRET_KEY'=''
  -e 'SITE_LOGO_URL'='PRIVATE...'
  -e 'TZ'='Europe/Berlin'
  -e 'INTEGRATIONS_ALLOW_PRIVATE_IPS'='True'
  -e 'PUID'='99'
  -e 'PGID'='100'
  -e 'UMASK'='022'
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.webui='http://[IP]:[PORT:8000]'
  -l net.unraid.docker.icon='https://miro.medium.com/v2/resize:fit:2400/1*m_ad1TBlJEKTn9n2sKZqgg.png'
  -p '8003:8000/tcp'
  -p '2525:2525/tcp'
  -v '/mnt/cache/appdata/monitoring/healthchecks':'/config':'rw'
  -e 'DEFAULT_FROM_EMAIL'='"Healthchecks" <example@org.net>'
  --health-cmd="curl
  --fail http://localhost:8000 || exit 1"
  --health-interval=1m
  --health-timeout=10s
  --health-retries=3 'lscr.io/linuxserver/healthchecks'
18e0a0376aad99ea1ca75107a08645f3c167526c36197ce88c7acf7b8c3c0d34
```

Before  `-e 'DEFAULT_FROM_EMAIL'='"Healthchecks" <example@org.net>'` was more like ones above when I enter them directly in unRAID.

**Reply by cuu508** [reply]
You are using a 3rd-party Docker image. Can you reproduce the issue with the official image from https://hub.docker.com/r/healthchecks/healthchecks ?

**Reply by saschabrockel** [reply]
I need a new instance to try. I tried to use your image with my current setup and it instantly broke. Asked for SECRET_KEY and I gave it some but after that, it said the database was corrupt. But it is not. Weird. I thought it should be the same.

**Reply by saschabrockel** [reply]
Okay, @cuu508 you were right. The behavior is very strange. With the original image, it works. But now I cannot send it to the mail address I used before. It's the same I used to send the mail. So I send it from info@xyz.com to info@xyz.com. No test notification ever appears even though it says "Test notification sent!". If I change it to Gmail for example it works just fine. Also with the correct name now. Why does this happen? It also does not require any confirmation if I change it to the info@xyz.com mail. When I change it to the Gmail then it will always ask for confirmation. info@xyz.com is also the email address of my Healthchecks account.

Is it a legacy problem because I used the other image or what? 

Another thing I really don't like about the original image is that cURL and/or wget are missing. I cannot check if the container is healthy with a simple health check like `--health-cmd="curl --fail http://localhost:8000 || exit 1" --health-interval=1m --health-timeout=10s --health-retries=3`. Can we not just add one of these packages?

**Reply by cuu508** [reply, marked as answer]
> So I send it from [info@xyz.com](mailto:info@xyz.com) to [info@xyz.com](mailto:info@xyz.com). No test notification ever appears even though it says "Test notification sent!".

Does it work if you try to send with the same SMTP credentials, to the same recipient, but with a different SMTP client?

> Another thing I really don't like about the original image is that cURL and/or wget are missing.

`http://localhost:8000/api/v3/status` would be a better endpoint for health checking. It runs a test query on the database and returns non-200 response if the query fails. With the current image you could make the HTTP request using python's urrlib.request, something like:

```
python -c "from urllib.request import urlopen; urlopen('http://localhost:8000/api/v3/status')"
```

**Reply by saschabrockel** [reply]
@cuu508 sorry it was not the fault of your app. Cloudflare / Gmail blocked my mails for a while.

Thank you for the healthcheck advise. I will try!

**Reply by cuu508** [reply]
I added a HEALTHCHECK instruction in the Dockerfile here: https://github.com/healthchecks/healthchecks/commit/8fed685f129f1eb1d66ca001f7e4872ac5f2f04d

So, after the next release, you will not need to add your own `--health-cmd` argument any more.
